### ISSUE TITLE
Data sorting dynamically by parameters send by user

### ISSUE THREAD

**Issue by yongkylie** [algolia/scout-extended#95]
Is it possible to do data sorting based on parameters being sent to Algolia?

For example, I have an e-commerce app that sell books. Sometimes I need to sort the data by popularity, sometimes by sales, sometimes by newest or oldest.

Can't find anything about this in the official Algolia docs.
Thanks!

**Comment by nunomaduro** [comment]
Every index has a unique sorting strategy, which is statically defined at the index level. Take a look at this docs and tell me if something is not clear for you: [https://www.algolia.com/doc/guides/ranking/sorting/](https://www.algolia.com/doc/guides/ranking/sorting/).

**Comment by yongkylie** [comment]
I see! I need to have multiple replicas of the index to achieve this. After I made the replica, then I could have different configuration of rankings for each replica.

Thanks!

**Comment by yongkylie** [comment]
Thank you for the reply @nunomaduro! After I read the articles, I am now confused where should I initiate `setSetting()` method on (as shown here: https://www.algolia.com/doc/guides/managing-results/refine-results/sorting/how-to/sort-an-index-by-date/?language=php#creating-a-replica). Is it alongside with my model?

My current state is, I had an index that works well without any sorting. I have created two replicas called `products_price_asc` and `products_price_desc`.

My code:
```
import App\Models\Product;
...
// Tried Product::setSettings(['replicas' => ['products_price_asc', 'products_price_desc']]) and not working
$result = Product::search($search_term);

return $result;
```

Thanks!

**Comment by nunomaduro** [comment]
No worries, I am here for you.

To search on the replica you need to use the `within` method:
```
$products = Product::search($searchTerm)
                ->within('products_price_asc')
                ->get();
```

Tell me if you need something else.

**Comment by pdmshrestha** [comment]
Hi, How to set settings for replica. 

Configuration
`
// conifg/scout-articles.php
<?php
return [
    'searchableAttributes' => [ 'title',  'content', ],
    'customRanking' => ['desc(published_at)'],
    'removeStopWords' => null,
    'disableTypoToleranceOnAttributes' => ['title', 'description'],
    'attributesForFaceting' => null,
    'unretrievableAttributes' => null,
    'ignorePlurals' => null,
    'queryLanguages' => null,
    'distinct' => null,
    'attributeForDistinct' => null,
    'replicas' => ['articles_asc', 'articles_desc'],
];
`
I need to configure the the replicas to sort by `created_at`.

**Comment by yongkylie** [comment]
@pdmshrestha 

1. Log in to your account at Algolia website.
2. Go to Indices menu.
3. If you haven't have a replica, go create one by pressing New > Replica beneath indices title.
4. Go to the replica indices then click on Configuration sub-menu.
5. Click Ranking and Sorting, and add a new searchable attribute (which is created_at).

**Comment by pdmshrestha** [comment]
@yongkylie Yes, I know that how to do and I have done that. But I was asking how to do that locally and sync with `php artisan scout:sync` to sync with remote configuration. For default index the settings are synced properly but I have issues with replicas only.

**Comment by nunomaduro** [comment]
@pdmshrestha Undertood your issue, can you create an issue with the feature request plz?
